### ISSUE TITLE
Unhandled rejection error on npm uninstall/remove

### ISSUE THREAD

**Issue by bubblecup** [npm/npm#17503]
#### I'm opening this issue because:

  - [x] npm is crashing.
 

#### What's going wrong?

On `npm remove`, I get an `Unhandled rejection Error: Argument #2: Expected string but got null` with the following stack trace:

```at addDependency (/usr/local/lib/node_modules/npm/lib/install/deps.js:515:3)
    at /usr/local/lib/node_modules/npm/lib/install/deps.js:463:5
    at /usr/local/lib/node_modules/npm/node_modules/slide/lib/async-map.js:52:35
    at Array.forEach (native)
    at /usr/local/lib/node_modules/npm/node_modules/slide/lib/async-map.js:52:11
    at Array.forEach (native)
    at asyncMap (/usr/local/lib/node_modules/npm/node_modules/slide/lib/async-map.js:51:8)
    at loadDeps (/usr/local/lib/node_modules/npm/lib/install/deps.js:455:3)
    at Array.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/slide/lib/bind-actor.js:15:8)
    at LOOP (/usr/local/lib/node_modules/npm/node_modules/slide/lib/chain.js:15:14)
    at /usr/local/lib/node_modules/npm/node_modules/slide/lib/chain.js:18:7
    at /usr/local/lib/node_modules/npm/lib/install/and-finish-tracker.js:8:8
    at resolveBranchDeps.then (/usr/local/lib/node_modules/npm/lib/install/deps.js:419:11)
    at tryCatcher (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:569:18)

npm ERR! cb() never called!

npm ERR! This is an error with npm itself. Please report this error at:
npm ERR!     <https://github.com/npm/npm/issues>
```

### supporting information:

 - `npm -v` prints: `5.0.4`
 - `node -v` prints: `8.1.2`
 - `npm config get registry` prints: `https://registry.npmjs.org/`
 - Windows, OS X/macOS, or Linux?: OS X
 - Network issues:
   - Geographic location where npm was run: Canada
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by hauntingEcho** [comment]
possibly related to #17497?  This is the same error I receive for `npm uninstall -g`

**Comment by sjones6** [comment]
I just received the same error:

`npm uninstall -g babel` resulted in the following error:

Unhandled rejection Error: Argument #2: Expected string but got nullbd522cd27b13
    at addDependency (/usr/local/lib/node_modules/npm/lib/install/deps.js:515:3)
    at /usr/local/lib/node_modules/npm/lib/install/deps.js:463:5
    at /usr/local/lib/node_modules/npm/node_modules/slide/lib/async-map.js:52:35
    at Array.forEach (native)
    at /usr/local/lib/node_modules/npm/node_modules/slide/lib/async-map.js:52:11
    at Array.forEach (native)
    at asyncMap (/usr/local/lib/node_modules/npm/node_modules/slide/lib/async-map.js:51:8)
    at loadDeps (/usr/local/lib/node_modules/npm/lib/install/deps.js:455:3)
    at Array.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/slide/lib/bind-actor.js:15:8)
    at LOOP (/usr/local/lib/node_modules/npm/node_modules/slide/lib/chain.js:15:14)
    at /usr/local/lib/node_modules/npm/node_modules/slide/lib/chain.js:18:7
    at /usr/local/lib/node_modules/npm/lib/install/and-finish-tracker.js:8:8
    at resolveBranchDeps.then (/usr/local/lib/node_modules/npm/lib/install/deps.js:419:11)
    at tryCatcher (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:569:18)

NPM: 5.0.3;
Node: 8.1.3

Appears to be the same as the others.

**Comment by hauntingEcho** [comment]
@sjones6 - try updating npm.  5.1.0 included a fix that solved the issue for me (#17590)

**Comment by kenany** [comment]
Yeah, I do believe this is resolved in `npm@5.1.0`. Could you give it a shot, @bubblecup?

**Comment by bubblecup** [comment]
Works great, thank you... and closed.
